### ISSUE TITLE
[Translation: Japanese] New JP translation work for Wiki page.

### ISSUE THREAD

**Issue by nicky008** [KC3Kai/KC3Kai#800]
I received a number of emails about the suggestion in localizing the Wiki page since began my first job in translating here but I have no time much in doing it yet.

And finally today I met one helper who want to do the localization works for JP.

I think it's time to realize this and it's the good stepping up for foreign users about better understanding in using this KC3Kai further than having a reference from the main English version only.

PS. And... welcome @nekoworkshop, a new member in the Japan translating section here.

**Comment by nekoworkshop** [comment]
Thanks Nicky. I didn't even have the courage to say hello in my first pull request.

As for the wiki. What I was talking about in my emails was actually the English wiki which will be the fundation if we were to make wiki in other languages. Unfortunately the progress of our wiki seems a bit late compared to the software.

![image](https://cloud.githubusercontent.com/assets/1346257/9769065/ba34d896-56f5-11e5-9a75-0ecd815c4e2b.png)

As @dragonjet pointed out, there *are* users who are having diffculties understanding the settings.

>ah just a heads up, I'm also planning to change the settings page for a bit. There had been multiple instances that users ask what this is, what it does, and one user was even frustrated he was like "it's like you want us to guess what each of these does" or something like that. Well it's true, and we need tooltips or description for each setting, apart from the wiki, where we do not have time to update xD

Here might be a good example of a guess-what-do-i-do setting.

![image](https://cloud.githubusercontent.com/assets/1346257/9769100/077c1bd2-56f6-11e5-8f2a-907c4409d7eb.png)

Simplified Chinese, Traditional Chinese and French transaltors included the word "preview" ( in zh-CN it's "spoiler", precisely speaking) to convey the idea of "letting user know what he got before the game does so". I guess that's the best we can do when the descriptions are limited to a few words with no tooltips nor wiki pages to supplement them.

To sum it up. I believe it would be better for us to focus on improving the wiki as well as the UIs rather than opening new fronts. Here is just my two cents and it's not my intention to push or hold back anyone. With so many brilliant people in this project currently I don't think my rusty coding skills would be needed but at least please allow me to share my knowledge of Japanese. Cheers (・ω・)bｸﾞｯ！

**Comment by bamboo3250** [comment]
パンパカパン @nekoworkshop Welcome to KC3Kai! Any help or suggestion will be appreciated here. Nice to work with you though ＼（＾▽＾）／

**Comment by dragonjet** [comment]
Welcome~ you first might wanna put your links on #142 so we can add you on devlist. I've also added you to the translator team on GitHub~

**Comment by dragonjet** [comment]
hm also, we should probably move the wiki to a `doc` repo which contains all markdown files instead of multiple languages on the same wiki, or if anyone knows a better way to do it, please let us know.

**Comment by nicky008** [comment]
IMO, I prefer `doc` repo for easier in complex contents management and looking nicer than putting in the same wiki.

**Comment by dragonjet** [comment]
there's one more option, in-app wiki, where all the articles, faqs, and help explanations are moved to another "Help" button on the menu.

I recommend the in-app help though - because not everyone stumbles upon this github link, and I see many links on sites directing towards the webstore installer, and those people never got to see the github, not even the dev blog.

**Comment by nicky008** [comment]
Hmm, in-app is not bad though when we considered in outsider people also.

**Comment by ReiFan49** [comment]
it means,
1. turn hardcoded faq into json/sofcoded faq
2. wiki mostly for developer guide
right?

**Comment by dragonjet** [comment]
yes, this is what I thought while redesigning the settings page. and right now it's still on adobe but About page and Settings page are merged and are tabified just like strategy room, and with this menu, we can also include help pages at the bottom

http://puu.sh/k6d2A.png

**Comment by dragonjet** [comment]
TBH idk anymore, maybe we should merge this to strategy room as well. true the hardest part of development is not coding itself but figuring out the UX :v

**Comment by nicky008** [comment]
Speak frankly, my UX idea in my head have been keeping hit my head inside since yesterday...

**Comment by dragonjet** [comment]
fire away

**Comment by ethrundr** [comment]
even though the thread is for JP Wiki translation, I'd wish to join up for the ID team, if we are going to include those wiki as in-app guides.

also, putting the in-app guide (or can be also called as FAQ section, imo) is necessary I think, especially for crucial and common bugs/issues.

**Comment by dragonjet** [comment]
yes @ethrundr we are the funnel of questions on social media, sometimes I just don't wanna open my Facebook anymore :laughing:

**Comment by ethrundr** [comment]
anyway @dragonjet, how about make a new thread (or turn this one) as the global thread to accommodate the Wiki translations work?

**Comment by dragonjet** [comment]
i think we can use this as the thread for the whole thing since the workflow will also change for the wiki translations

**Comment by dragonjet** [comment]
how about like this?

![20 0](https://cloud.githubusercontent.com/assets/557312/9783299/dee6210a-57d5-11e5-8f8f-1e87449a6eee.png)

**Comment by dragonjet** [comment]
new thread created: #810

**Comment by nicky008** [comment]
thx @dragonjet for a further more consolidated thread... :3

**Comment by dragonjet** [comment]
alright, going back to the topic of wiki, before you guys start the translation for JP and ID wikis, we should first finalize the `en` :laughing: We should probably do a documentation week where we just write FAQ instead of code :v

**Comment by ethrundr** [comment]
it's a good idea.
btw @dragonjet, just to keep track for what information still missing, why won't we make a checklist here?

**Comment by dragonjet** [comment]
sure, though will move the redesign to next week v20, including the settings page at #773

**Comment by ethrundr** [comment]
I'll try help as much as I can do within my empty times during work, but don't give too high a hope, I don't want to break people heart XD

**Comment by dragonjet** [comment]
**EN**
:black_medium_square: a
:black_medium_square: b
:black_medium_square: c

**JP** (*on hold until `en` is finalized*)
:black_medium_square: a
:black_medium_square: b
:black_medium_square: c

**ID** (*on hold until `en` is finalized*)
:black_medium_square: a
:black_medium_square: b
:black_medium_square: c

**Comment by nekoworkshop** [comment]
@dragonjet Just a head up that the wiki portal (https://github.com/KC3Kai/KC3Kai/wiki) is a little bit outdated compared to the sidebar. It's missing a few parts, including the development build guide ( which took me couple seconds to figure out...)

**Comment by dragonjet** [comment]
@nekoworkshop yep, though we should use the sections above when we rewrite, the sidebar and actual content right now are outdated, like it still contains DMM frame etc which do not apply anymore

**Comment by Crazyteitoku** [comment]
Ok I know this issue is a bit old but I have sort of did a bit of update with the information on the sidebar.

What ideally needs to happen is that one is able to sync a copy of what we have on https://github.com/KC3Kai/KC3Kai/wiki to the new dedicated home of https://github.com/KC3Kai/kc3-docs especially for the En part. This way at least a sort of workable basis has been ratified. From there on we need to form groups to focus on documentation as well as translation.

**Comment by dragonjet** [comment]
yea @Crazyteitoku I kinda regret writing some articles in markdown :(

**Comment by Crazyteitoku** [comment]
@dragonjet no problem with that, it seems as though the markdown language is just as extensible. :sweat_smile: 

What needs to happen though is how one contributes to that section really. The only way I can think of is to write the new documentation using markdown language and then do git push to that repository.

Another problem would then be how will these new documentation as well as presentation be visible to the user especially if and when they want to read the accompanying documentation and that they clicked the help button from within the viewer?

**Comment by dragonjet** [comment]
wait @Crazyteitoku it's the opposite, I mean I regret writing it on MediaWiki XD
If all articles are markdown we can easily copy-paste all of them into the doc repo

**Comment by Crazyteitoku** [comment]
Ah yeah I knew :laughing: but that in my opinion is a really small issue. The bigger picture is how to now address the new documentation. I am somewhat flexible with both languages so that isn't so much of a problem as per se. It is really just now how to write up new changes so that users can easily read and reference it up as needed.

**Comment by Javran** [comment]
just giving some ideas:

* let me introduce [pandoc](http://pandoc.org/README.html): an universal document format converter that supports both markdown and mediawiki (and written in my favorite programming language XD). I can do the conversion in my spare time.

* can [GitHub Pages](https://pages.github.com/) be utilized for keeping documents? we can stil edit documents in https://github.com/KC3Kai/kc3-docs but output well-formatted HTML pages for users. but definitely some extra tasks like setting up Jekyll / Hakyll needs to be done.

**Comment by dragonjet** [comment]
whoa I'm all new to these things :3

I think we can manual the conversion. The good thing is that we also need to go over the contents since alot have changed since it was written.

hm might look at github pages, but im not sure if it support multi-language though. that's the reason why we moved to doc-repo :v

**Comment by Javran** [comment]
@dragonjet I think that's also fine if most of docs are out-dated.

for github pages, think it as a free static webpage hosting service, all a static website can do can be done in github pages. After all it's purpose is for developers to keep project sites.

usually the workflow is:

1. config a static website generator ([jekyll](https://jekyllrb.com/) is the most popular choice)
2. write posts in plain text / markdown / etc.
3. run the generator so you'll have a bunch of generated html pages organized in the standard way (index.html for example)
4. push them to the website repo

just investigated about jekyll, it does have many ways of supporting multiple languages,
one example is https://www.sylvaindurand.org/making-jekyll-multilingual/ 
switch between `en` and `fr` in the top bar to see it in action

**Comment by dragonjet** [comment]
oh that's nice, I even saw an automatic page generator on github, dunno what that does though xD

**Comment by Crazyteitoku** [comment]
Actually this would potentially save a lot of elbow grease. :+1: 

So basically we have unanimously to convert all the wiki pages into html format for use on the kc3kai-docs project? :laughing:

**Comment by Javran** [comment]
@Crazyteitoku you can write it in markdown, the html conversion is handled by the static webpage generator. (I thnk they also supports mediawiki)

**Comment by Crazyteitoku** [comment]
@Javran ~~yeah but what I meant was that the documentation I saw for instance on [kc3kai-docs] (https://github.com/KC3Kai/kc3-docs/tree/master/en) seems to be .md file? the links are pointing to an non-existent area within that project.~~ I derped :fearful: disregard what I said: .md **is** markdown :laughing: I think I am slowly getting the hang of this. :sweat_smile:

**Comment by Javran** [comment]
@Crazyteitoku that's fine, both `.markdown` and `.md` are vaild.

**Comment by Crazyteitoku** [comment]
Ok following @Javran suggestion on [pandoc](http://pandoc.org/README.html), I have begun conversion of mediawiki format for documentations in favour of markdown format for use on kc3-docs part of the project page.

As a result I am now self-assigning to this issue and have begun revising documentation as needed. Several now converted markdown files will need updating to reflect on the ever changing KC3 Kai and whilst I have not pushed any newly converted markdown formats, I hereby ask if anyone wishes to also contribute please come forward so that I will know if there's any need to push other pages first.

For a WYSIWYG (What You See Is What You Get) editor which works on a fair few formats as well as platforms I am using [Haroopad](http://pad.haroopress.com/) following the suggestion from [linuxbsdos.com](http://linuxbsdos.com/2014/10/05/the-best-markdown-editors-for-linux/).

**Comment by Crazyteitoku** [comment]
**Initial stage - Planning:**
:white_check_mark: conversion software from mediawiki to markdown
:white_check_mark: "IDE" for markdown
:white_check_mark: filename conventions
:black_medium_square: folder/directory structure
:black_medium_square: discussion of possible inclusion of images for the wiki
:black_medium_square: document template 

**Stage 2 - migration**
:white_check_mark: git clone
:black_medium_square: updating and pushing conversions
:black_medium_square: navigation issues between wiki pages.
:black_medium_square: unify the templates for documentation into a suitable and agreed layout.
:black_medium_square: translation?

**Final stage - cleanup and resumption of normal activities**
:black_medium_square: removal of KC3Kai.wiki and link readers to the new project area.
:black_medium_square: Update wiki pages as per necessary.
